src/components/InstructorReviewPage.js

```diff
@@ -0,0 +1,345 @@
+      this.props.history.push('/instructorpage')
+    } catch (e) {
+      console.log('error happened', e)
+      this.props.setError(e.response.data.error)
+    }
+  }
+
+  render() {
+    const {
+      answerSheet,
+      updateAnswer,
+      submittedReview,
+      groups,
+      selectedGroup,
+      selectGroup,
+      initializeAnswerSheet
+    } = this.props
+
+    if (submittedReview === true) {
+      return (
+        <div>
+          <h3>You have reviewed every group you are instructing.</h3>
+        </div>
+      )
+    } else if (answerSheet && groups.length > 0) {
+      return (
+        <div>
+          <ConfigurationSelectWrapper label="Select group">
+            <ConfigurationSelect
+              selectedGroup={selectedGroup}
+              groups={groups}
+              groupSelectHandler={groupSelectHandler}
+              selectGroup={selectGroup}
+              fetchInstructorReviewQuestions={
+                this.fetchInstructorReviewQuestions
+              }
+              initializeAnswerSheet={initializeAnswerSheet}
+            />
+          </ConfigurationSelectWrapper>
+          <h1>{groups[selectedGroup].groupName}</h1>
+
+          <Reviews answerSheet={answerSheet} updateAnswer={updateAnswer} />
+          <Button
+            margin-right="auto"
+            margin-left="auto"
+            variant="contained"
+            color="primary"
+            onClick={(event) =>
+              this.Submit(
+                event,
+                answerSheet,
+                groups[selectedGroup].groupName,
+                groups[selectedGroup].id
+              )
+            }
+          >
+            Submit
+          </Button>
+        </div>
+      )
+    } else {
+      return (
+        <div>
+          <p>Loading.</p>
+        </div>
+      )
+    }
+  }
+}
+
+const groupSelectHandler = (
+  value,
+  selectGroup,
+  fetchInstructorReviewQuestions,
+  groups,
+  initializeAnswerSheet
+) => {
+  if (
+    !window.confirm('Changing group will wipe current inserted information.')
+  ) {
+    return
+  }
+
+  selectGroup(value)
+  fetchInstructorReviewQuestions(
+    groups[value].students,
+    questionsJson,
+    initializeAnswerSheet
+  )
+}
+const ConfigurationSelectWrapper = ({ label, children }) => (
+  <div style={{ padding: 20 }}>
+    <Typography variant="caption">{label}</Typography>
+    {children}
+  </div>
+)
+
+const ConfigurationSelect = ({
+  selectedGroup,
+  groupSelectHandler,
+  groups,
+  fetchInstructorReviewQuestions,
+  selectGroup,
+  initializeAnswerSheet
+}) => {
+  return (
+    <Select
+      value={selectedGroup}
+      onChange={(e) =>
+        groupSelectHandler(
+          e.target.value,
+          selectGroup,
+          fetchInstructorReviewQuestions,
+          groups,
+          initializeAnswerSheet
+        )
+      }
+    >
+      {groups.map((group, index) => (
+        <MenuItem key={index} value={index} data-cy="group">
```

Comment:
data-cy is same in all menu items, will cause problems with testing.

Reply:
Removed data-cy fields.